### DISCUSSION TITLE
Remove cortex and thehive after update to 120

### DISCUSSION THREAD

**Discussion by tomjones1977** [Security-Onion-Solutions/securityonion#7870] [Unsupported Versions] [answered]
After updating a distributed system from 110 to 120 the manager node shows a fault in Grid.

Running `so-status` shows that `so-cortex`, `so-thehive`, and `so-thehive-es` are `[ MISSING ]` which is expected as they have been replaced by Cases.  How can I remove these without impacting on the system?

**Comment by dougburks** [comment, marked as answer]
I just tested upgrading from 2.3.91 to 2.3.120 and was not able to duplicate this issue.

soup should automatically remove these services from so-status.conf:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/soup#L507
> sed -i '/so-thehive-es/d;/so-thehive/d;/so-cortex/d' /opt/so/conf/so-status/so-status.conf

Check your soup log to see if there is any evidence for this command or anything else failing.

**Reply by tomjones1977** [reply]
Odd - I can't see anything in the soup log that would indicate a failure, and the conf file still refers to those containers.  I've run the above command manually which has cleared the so-status error as expected, but is there anything else that should be checked manually?

**Reply by defensivedepth** [reply]
In the soup log, do you see the following being echoed:

echo "Post Processing for 2.3.120"

Do you see any errors happening around that timeframe?

**Reply by dougburks** [reply]
I suspect your box is probably OK and this was just a simple cosmetic issue. Seems improbable but I'm wondering if an old salt process was still running and added those entries back to so-status.conf after soup removed them. Do you happen to know if they were the last 3 lines in the file?

**Reply by wdhachfeld** [reply]
Just an FYI for anyone finding this in a search...

I ran into this myself upgrading from 2.3.110 to 2.3.120 on a standalone installation. Last night I had logged in via ssh and kicked off soup. When I checked everything this morning my ssh session had been terminated and the last line on the screen was:

> Running a highstate. This could take several minutes.

Corresponding to [this](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/soup#L1212) line in the soup script. My soup log file also shows this as the last printed line as well. Since [this](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/soup#L1224) line was never reached, the post-upgrade changes [here](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/soup#L1230) were not executed and consequently the `so-cortex`, `so-thehive`, and `so-thehive-es` containers still appeared in my `so-status`.

Everything else seemed functional. So I manually removed these containers from `so-status.conf` via the `sed` command above that @dougburks  referenced. The status now appears fine and everything looks to be working.

**Reply by dougburks** [reply]
> Last night I had logged in via ssh and kicked off soup. When I checked everything this morning my ssh session had been terminated

Yes, if the ssh session terminates then any processes running in that session would terminate. In the future, you might want to avoid leaving it unattended especially if the machine you are ssh'ing from is configured to sleep after a period of time. You might also consider using something like screen or tmux so that if your ssh session terminates, the processes will continue running on the server.
